### PR TITLE
Auto OCP-38132

### PR DESCRIPTION
**Pull request by huiran0826**
Auto OCP-38132

Logs: https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/Runner-v3-smoke/2965/consoleFull

@zhaozhanqi @anuragthehatter @weliang1 @rbbratta @asood @brahaney @kedark3

### CODE REVIEW

File: features/step_definitions/networking.rb [hunk 1]
```diff
@@ -1292,3 +1292,60 @@
...
+  else
+    ovn_cmd=%w(hostname -f)
+  end
+
+  ovn_pod = BushSlicer::Pod.get_labeled("app=ovnkube-node", project: project("openshift-ovn-kubernetes", switch: false), user: admin) { |pod, hash|
+  pod.node_name == node.name
+}.first
+  @result=ovn_pod.exec(*ovn_cmd, as: admin, container: "ovnkube-node")
+  if @result[:success]
+     hostname_result=@result[:response].lines.first.strip
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:1328]: same, spaces around ` = ` please.

File: features/step_definitions/networking.rb [hunk 2]
```diff
@@ -1292,3 +1292,60 @@
...
+  cb[cb_type_hostname] = hostname_result
+  logger.info "nodes's corresponding #{type_name} hostname is stored in the #{cb_type_hostname}"
+end
+
+Given /^I set#{OPT_QUOTED} hostname to external ids on the "([^"]*)" node$/ do |custom_hostname,node_name|
+  ensure_admin_tagged
+
+  # set hostname to external ids
+  ovsvsctl_cmd = %w(ovs-vsctl set open .)
+  external_hostname="external_ids:hostname=#{custom_hostname}"
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:1340]: same, spaces around ` = ` please.

File: features/step_definitions/networking.rb [hunk 3]
```diff
@@ -1292,3 +1292,60 @@
...
+  cb[cb_ovn_hostname] = ovn_state_output.split("hostname=").last.split(", ovn-bridge-mappings").first
+  logger.info "nodes's corresponding hostname from external_ids is stored in the #{cb_ovn_hostname} clipboard."
+end
+
+Given /^I store the#{OPT_QUOTED} hostname in the#{OPT_SYM} clipboard for the "([^"]*)" node$/ do |type_name,cb_type_hostname,node_name|
+  ensure_admin_tagged
+  cb_type_hostname ||= "type_hostname"
+  case type_name
+  when "short"
+    ovn_cmd=%w(hostname -s)
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:1318]: we usually have a space around ` = ` ? I'm assuming that is Ruby style.

`    ovn_cmd = %w(hostname -s)`

File: features/step_definitions/networking.rb [hunk 4]
```diff
@@ -1292,3 +1292,60 @@
...
+  else
+    ovn_cmd=%w(hostname -f)
```
Comment (rbbratta) [inline features/step_definitions/networking.rb:1320]: same, spaces around ` = ` please.

File: features/step_definitions/networking.rb (hunk 1 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: fixed it, thanks!

File: features/step_definitions/networking.rb (hunk 2 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: fixed it, thanks!

File: features/step_definitions/networking.rb (hunk 3 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: fixed it, thanks!

File: features/step_definitions/networking.rb (hunk 4 above)
Comment (huiran0826) [reply features/step_definitions/networking.rb]: fixed it, thanks!

File: features/step_definitions/networking.rb [hunk 5]
```diff
@@ -1292,3 +1292,60 @@
   default_network = network_operator.default_network(user: admin)
   raise "env doesn't have IPSec enabled" unless default_network["ipsecConfig"]
 end
+
+Given /^I store the hostname from external ids in the#{OPT_SYM} clipboard on the "([^"]*)" node$/ do |cb_ovn_hostname, node_name|
+  ensure_admin_tagged
+  cb_ovn_hostname ||= "ovn_hostname"
+
+  ovsvsctl_cmd = %w(ovs-vsctl list open .)
+  ovn_pod = BushSlicer::Pod.get_labeled("app=ovnkube-node", project: project("openshift-ovn-kubernetes", switch: false), user: admin) { |pod, hash|
```
Comment (pruan-rht) [inline features/step_definitions/networking.rb:1301]: indentation... Also, you can call step %Q/a pod becomes ready with label.../

Comment (huiran0826) [reply features/step_definitions/networking.rb:1302]: Thanks, fixed indentation. But this step is required for a ready pod on a specific node, looks like %Q/a pod becomes ready with label.../ doesn't have the node specification. So keep the original node here.

File: features/networking/ovn.feature [hunk 6]
```diff
@@ -465,3 +465,61 @@ Feature: OVN related networking scenarios
...
+  Given I store the ovnkube-master "north" leader pod in the clipboard
+  Given I wait up to 15 seconds for the steps to pass:
+  """
+  And admin executes on the pod "northd" container:
+    | bash | -c | ovn-sbctl --no-leader-only list chassis \|grep "<%= cb.short_hostname %>" |
+  Then the step should succeed
+  And the output should contain:
+    | <%= cb.short_hostname %> |
+  And the output should not contain:
+    | <%= cb.original_hostname%> |
```
Comment (rbbratta) [inline features/networking/ovn.feature:501]: Doesn't seem to work on vSphere.  The external hostname is the same as the shortname.

```
external_ids        : {hostname=control-plane-0, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.16", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="233f3038-086f-4a2e-8e03-1601e7f08a78"}
external_ids        : {hostname=compute-1, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.54", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="1e994f24-bc20-43d7-804d-ef88e1ccd591"}
external_ids        : {hostname=control-plane-1, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.86", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="1f28920a-e538-42e3-8777-c8170e181f15"}
external_ids        : {hostname=compute-0, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.19", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="c2e44006-4437-4cfe-9c78-a090bd060087"}
external_ids        : {hostname=control-plane-2, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.48", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="cf3944a9-03b4-4ff3-89fe-da5c4d04a68b"}
external_ids        : {hostname=compute-2, ovn-bridge-mappings="physnet:br-ex", ovn-encap-ip="172.31.248.74", ovn-encap-type=geneve, ovn-monitor-all="true", ovn-openflow-probe-interval="180", ovn-remote="ssl:172.31.248.16:9642,ssl:172.31.248.48:9642,ssl:172.31.248.86:9642", ovn-remote-probe-interval="30000", rundir="/var/run/openvswitch", system-id="402d2f1e-b021-45e1-b4b9-f1f9514e2c91"}
```

Comment (rbbratta) [reply features/networking/ovn.feature:501]:
```
oc exec ovnkube-master-pm274  --kubeconfig=ocp4_admin.kubeconfig -n openshift-ovn-kubernetes  --container=northd -i -- bash -c ovn-sbctl\ --no-leader-only\ list\ chassis\ \|grep\ \"compute-0\"

hostname            : compute-0

STDERR:
E0420 08:14:14.136111 3394046 v2.go:105] write tcp 172.16.16.104:54570->54.189.75.126:3128: write: broken pipe


[12:14:14] INFO> Exit Status: 0

RuntimeError: pattern found: compute-0
```

Comment (huiran0826) [reply features/networking/ovn.feature:501]: Thanks Ross for reviewing it. Actually this case is from bug https://bugzilla.redhat.com/show_bug.cgi?id=1908921 and from comments https://bugzilla.redhat.com/show_bug.cgi?id=1908921#c17, we can see the issue was reproduced on AWS and we also followed that suggestion to verify the bug.  I have filtered that case in Polarion for AWS platform.

File: features/networking/ovn.feature [hunk 7]
```diff
@@ -465,3 +465,61 @@ Feature: OVN related networking scenarios
...
+  When I set "<%= cb.original_hostname%>" hostname to external ids on the "<%= cb.nodes[0].name %>" node
+  Then the step should succeed
+  """
+
+  # Check short hostname was set
+  Given I store the ovnkube-master "north" leader pod in the clipboard
+  Given I wait up to 15 seconds for the steps to pass:
+  """
+  And admin executes on the pod "northd" container:
+    | bash | -c | ovn-sbctl --no-leader-only list chassis \|grep "<%= cb.short_hostname %>" |
```
Comment (rbbratta) [inline features/networking/ovn.feature:496]: if we want exact we could use `find`   This is more what the OVN code itself uses.

`| ovn-sbctl | --no-leader-only | --format=csv | --data=bare  |  --no-heading  |  --columns=hostname | find | chassis | hostname=<%= cb.short_hostname %> |`

File: features/networking/ovn.feature [hunk 8]
```diff
@@ -465,3 +465,61 @@ Feature: OVN related networking scenarios
     Then the step should succeed
     # the iface-id should be quoted if it starts with a digit
     And the output should contain "<%= project.name %>"
+
+
+# @author huirwang@redhat.com
+# @case_id OCP-38132
+@admin
+@destructive
+Scenario: Should no intermittent packet drop from pod to pod after change hostname
```
Comment (liangxia) [inline features/networking/ovn.feature:474]: All those need one more level indention. You can compare with existing ones from above.

Comment (huiran0826) [reply features/networking/ovn.feature]: @liangxia  Thanks, fixed it.

**Comment by huiran0826** [comment]
Comments have been resolved. 
/assign @liangxia

**Comment by liangxia** [comment]
There are conflicts that's blocking the merge process.

**Pull request merged**